controller/src/main/java/io/pravega/controller/store/stream/PravegaTablesStreamMetadataStore.java

```diff
@@ -73,19 +76,25 @@
         this.completedTxnGC = new ZKGarbageCollector(COMPLETED_TXN_GC_NAME, zkStoreHelper, this::gcCompletedTxn, gcPeriod);
         this.completedTxnGC.startAsync();
         this.completedTxnGC.awaitRunning();
+        this.completedTxnGCRef = new AtomicReference<>(completedTxnGC);
         this.counter = new ZkInt96Counter(zkStoreHelper);
         this.storeHelper = new PravegaTablesStoreHelper(segmentHelper, authHelper, executor);
         this.executor = executor;
     }
 
-    private CompletableFuture<Void> gcCompletedTxn() {
+    @VisibleForTesting 
+    CompletableFuture<Void> gcCompletedTxn() {
         List<String> batches = new ArrayList<>();
         return withCompletion(storeHelper.expectingDataNotFound(storeHelper.getAllKeys(COMPLETED_TRANSACTIONS_BATCHES_TABLE)
                                                          .collectRemaining(batches::add)
                                                          .thenApply(v -> {
                                                                      // retain latest two and delete remainder.
                                                                      if (batches.size() > 2) {
-                                                                         return batches.subList(0, batches.size() - 2);
+                                                                         List<String> list = batches
+                                                                                 .stream().sorted(Comparator.comparingLong(Long::parseLong))
+                                                                                 .collect(Collectors.toList());
+
+                                                                         return list.subList(0, batches.size() - 2);
```

Comment:
Why we only keep the 2 last batches? Is this an optimal value or it is just a value that sounds reasonable? In the latter case, we may want to replace that by a constant, so it can be easily modified if necessary.

Reply:
We keep the last two values because the latest batch is what is still being written to. 
And during the time when we transition from previous batch to new batch, callers may still be interested in transactions written to previous batch as the transactions written into it should be made available for at least the GC period. 
Everything before the latest two are guaranteed to be before the GC period so we can purge them.